### PR TITLE
Execution abstract first test

### PR DESCRIPTION
**Pull request by hungneox**
* Change \ArrayObject to array for easier to debug
* First Abstract test

### CODE REVIEW

File: src/Execution/ExecutionStrategy.php [hunk 1]
```diff
@@ -189,7 +189,7 @@ private function doesFragmentConditionMatch(
 
         $conditionalType = typeFromAST($this->context->getSchema(), $typeConditionNode);
 
-        if ($conditionalType === $type) {
+        if (gettype($conditionalType) === gettype($type)) {
```
Comment (crisu83) [inline src/Execution/ExecutionStrategy.php]: I'm not a fan of this solution. Simply comparing the instances should work.

File: src/Execution/ExecutionStrategy.php [hunk 2]
```diff
@@ -767,12 +776,12 @@ private function collectAndExecuteSubFields(
...
                     $fieldNode->getSelectionSet(),
                     $subFields,
-                    new \ArrayObject()
+                    $visitedFragmentNames
                 );
             }
         }
 
-        if ($subFields->count()) {
+        if (count($subFields) > 0) {
```
Comment (crisu83) [inline src/Execution/ExecutionStrategy.php]: Can't we use `!empty($subFields)`?

File: src/Execution/ExecutionStrategy.php (hunk 1 above)
Comment (hungneox) [reply src/Execution/ExecutionStrategy.php]: The issue is It didn't work :<

File: src/Execution/ExecutionStrategy.php (hunk 2 above)
Comment (hungneox) [reply src/Execution/ExecutionStrategy.php]: `if(count($subFields))` also works. But this for clarity.

**Pull request merged**